src/ray/object_manager/object_buffer_pool.h

```diff
@@ -50,7 +51,8 @@ class ObjectBufferPool {
   /// \param store_socket_name The socket name of the store to which plasma clients
   /// connect.
   /// \param chunk_size The chunk size into which objects are to be split.
-  ObjectBufferPool(const std::string &store_socket_name, const uint64_t chunk_size);
+  ObjectBufferPool(std::shared_ptr<PullManager> pull_manager,
```

Comment:
Rather than passing in pull manager into the buffer pool class, would it be cleaner to intercept the chunk create requests at a higher layer? It seems unfortunate a low level buffer pool class is concerned with admission control.

Reply:
I moved it here to avoid a race condition where the request gets deactivated in between checking whether the object is still active or not. I think I can move it back though, and just check after creating the chunk whether the request is still active.

Reply:
+1 in this case, we don't need locks anymore too right?

Reply:
We still need a lock because the chunk receives are handled by other threads.